.projenrc.ts

```diff
@@ -69,6 +71,26 @@ class CustomTypescriptProject extends typescript.TypeScriptProject {
 
     writeFileSync(rootPackageJson, `${JSON.stringify(updated, null, 2)}\n`);
   }
+
+  public renderWorkflowSetup(
+    options?: RenderWorkflowSetupOptions | undefined
+  ): JobStep[] {
+    return [
+      ...super.renderWorkflowSetup(options),
+      // https://github.com/aws-actions/configure-aws-credentials#sample-iam-role-cloudformation-template
+      // the aws-org stacks create an OIDC provider for github.
+      {
+        name: "Configure AWS Credentials",
+        uses: "aws-actions/configure-aws-credentials@v1",
+        with: {
+          "role-to-assume":
+            "arn:aws:iam::593491530938:role/githubActionStack-githubactionroleA106E4DC-14SHKLVA61IN4",
```

Comment:
How does github actions authenticate to assume this role?

Reply:
OIDC uses github itself as the identity and in the role we assert which repos can assume the role.

```ts
    const provider = new OpenIdConnectProvider(this, "githubProvider", {
      url: "https://token.actions.githubusercontent.com",
      clientIds: ["sts.amazonaws.com"],
      thumbprints: ["6938fd4d98bab03faadb97b34396831e3780aea1"],
    });
    const principal = new OpenIdConnectPrincipal(provider, {
      StringLike: {
        "token.actions.githubusercontent.com:sub": "repo:functionless/*",
      },
    });
```

Reply:
That is cool!

Reply:
Did you deploy that OIDC policy with CloudFormation?